src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -204,20 +236,28 @@ abstract class AbstractCheckTask(
      * Task subscribe to messages stream with sequence after call.
      * This method should be called only once otherwise it throws IllegalStateException.
      * @param sequence message sequence from the previous task.
+     * @param checkpointTimestamp checkpoint timestamp from the previous task
      * @param executorService executor to schedule pipeline execution.
+     * @param untrusted flag is guarantee that the previous sequence data is correct
      * @throws IllegalStateException when method is called more than once.
      */
-    private fun begin(sequence: Long = DEFAULT_SEQUENCE, executorService: ExecutorService = createExecutorService()) {
+    private fun begin(sequence: Long = DEFAULT_SEQUENCE, checkpointTimestamp: Timestamp? = null,
+                      executorService: ExecutorService = createExecutorService(), untrusted: Boolean = false) {
         if (!taskState.compareAndSet(State.CREATED, State.BEGIN)) {
             throw IllegalStateException("Task $description already has been started")
         }
-        LOGGER.info("Check begin for session alias '{}' with sequence '{}' timeout '{}'", sessionKey, sequence, timeout)
+        LOGGER.info("Check begin for session alias '{}' with sequence '{}' and task timeout '{}'", sessionKey, sequence, taskTimeout)
         this.lastSequence = sequence
         this.executorService = executorService
+        if (untrusted && !isIgnoreUntrustedFlag()) {
+            doOnUntrustedExecution()
+            return
```

Comment:
Why don't we start the task when the untrusted checkpoint is passed?

Reply:
Refer to https://github.com/th2-net/th2-check1/pull/36#issuecomment-899487957